README.md
# React Material UI Carousel [![npm version](https://img.shields.io/npm/v/react-material-ui-carousel.svg?style=flat)](https://www.npmjs.com/package/react-material-ui-carousel)

## Description

A Generic, extendible Carousel UI component for React using [Material UI](https://material-ui.com/)  
It switches between given children using a smooth animation.  
Provides next and previous buttons.
Also provides interactible bullet indicators.

## Live Demo

Take a look at this interactible [Live Demo](https://learus.github.io/react-material-ui-carousel)

## Installation

```shell
npm install react-material-ui-carousel --save
```

**Note:**

You will need to have Material UI installed, in order to use this library/component

```shell
npm install @mui/material
npm install @mui/icons-material
npm install @mui/styles
```

### Other Versions

```shell
# Version 2 with MUI 4
npm install react-material-ui-carousel@v2 --save
npm install @material-ui/core
npm install @material-ui/icons


# Version 2 with MUI 5 support
npm install react-material-ui-carousel@v2mui5 --save
```

## Usage Example

```jsx
import React from 'react';
import Carousel from 'react-material-ui-carousel'
import { Paper, Button } from '@mui/material'

function Example(props)
{
    var items = [
        {
            name: "Random Name #1",
            description: "Probably the most random thing you have ever seen!"
        },
        {
            name: "Random Name #2",
            description: "Hello World!"
        }
    ]

    return (
        <Carousel>
            {
                items.map( (item, i) => <Item key={i} item={item} /> )
            }
        </Carousel>
    )
}

function Item(props)
{
    return (
        <Paper>
            <h2>{props.item.name}</h2>
            <p>{props.item.description}</p>

            <Button className="CheckButton">
                Check it out!
            </Button>
        </Paper>
    )
}
```

## Next & Prev Usage

```jsx
    <Carousel
        next={ (next, active) => console.log(`we left ${active}, and are now at ${next}`); }
        prev={ (prev, active) => console.log(`we left ${active}, and are now at ${prev}`); }
    >
        {...}
    </Carousel>

    // OR

    <Carousel
        next={ () => {/* Do stuff */} }
        prev={ () => {/* Do other stuff */} }
    >
        {...}
    </Carousel>

    // And so on...
```

Note: `onChange` works in a similar fashion. See [Props](#props) below.

## Customizing Navigation

### Navigation Buttons - Customizing the default solution

These are the props that are used to directly customize the Carousel's default buttons:

* NextIcon
* PrevIcon
* navButtonsProps
* navButtonsWrapperProps
* fullHeightHover

#### Example #1

Say we don't like the default icons used for the next and prev buttons
and want to change them to be an MUI Icon or an image of our own.

```jsx


import RandomIcon from '@@mui/icons-material/Random'; // Note: this doesn't exist

<Carousel
    NextIcon={<RandomIcon/>}
    PrevIcon={<RandomIcon/>}
    // OR
    NextIcon={<img src="http://random.com/next"/>}
    PrevIcon={<img src="http://random.com/prev"/>}
>
    {...}
</Carousel>
```

The `NextIcon` and `PrevIcon` is of type `ReactNode`, meaning it can be any JSX element or a string. *Note: Extra styling may be needed when using those props*.

#### Example #2

Let's now say we don't like the default graphite background of the buttons, nor do we like the fact that it is round.  
We also want to place them under the main Carousel, and finally remove the arrows and have "next" and "prev" accordingly to each button.

A very important note here, is that any styles specified by the user **DO NOT OVERRIDE THE EXISTING STYLES**. They work in tandem with them. That means, that if you want to change, or get rid of a CSS attribute you will have to override it or unset it. The [Default styles](#default-styles) are given at the end of this section, and are part of the code.

```jsx
<Carousel
    fullHeightHover={false}     // We want the nav buttons wrapper to only be as big as the button element is
    navButtonsProps={{          // Change the colors and radius of the actual buttons. THIS STYLES BOTH BUTTONS
        style: {
            backgroundColor: 'cornflowerblue',
            borderRadius: 0
        }
    }} 
    navButtonsWrapperProps={{   // Move the buttons to the bottom. Unsetting top here to override default style.
        style: {
            bottom: '0',
            top: 'unset'
        }
    }} 
    NextIcon='next'             // Change the "inside" of the next button to "next"
    PrevIcon='prev'             // Change the "inside of the prev button to "prev"
>
    {...}
</Carousel>
```

Of course, extra styling to the button wrappers, or indicators might be needed to achieve exactly what we may be looking for. *Note: You can also use `className` to change the styles externally*.

### Customizing the navigation buttons directly

Do directly customize/change the navigation buttons `NavButton` prop, that allows the user to take complete control of the components rendered as the navigation buttons. It should be used like this:

#### Example

```jsx
import {Button} from '@mui/material';

<Carousel
    NavButton={({onClick, className, style, next, prev}) => {
        // Other logic

        return (
            <Button onClick={onClick} className={className} style={style}>
                {next && "Next"}
                {prev && "Previous"}
            </Button>
        )
    }}
>
    {...}
</Carousel>
```

##### Parameters Explanation

* `onClick`: The function that handles actual navigation. If you do not add this to your component, the buttons will not work.
* `className`: The className given by the carousel component. This is used to handle Visible/Invisible, hover, and user specified styles (e.g. from navButtonProps). Apply it to the outmost element.
* `style`: The style given by the carousel component. Used to give any user specified styles (e.g. from navButtonProps).
* `next`: Boolean value that specifies whether this is the next button.
* `prev`: Boolean value that specifies whether this is the prev button.

The prop value must be a function that returns a component. All parameters are optional as far as styling goes (**not functionality**), but it is advised you use them as shown above.  
As implied, any `className`s or `style`s specified in the navButtonsProps will only be used iff you apply the given `className` and `style` parameters.

### Customizing the Indicators

There are 4 props that handle indicator customization

* IndicatorIcon
* activeIndicatorIconButtonProps
* indicatorIconButtonProps
* indicatorContainerProps

#### Example

Let's say we would like to change the indicator icon from a circle to a something else, for example a little house

```jsx
import Home from '@mui/icons-material/Home';

<Carousel
    IndicatorIcon={<Home/>}
    // OR
    IndicatorIcon={<img src="http://random.com/home"/>}
>
    {...}
</Carousel>
```

The `IndicatorIcon`  works the same way as the `NextIcon` and `PrevIcon` prop.


#### Example #2

Let's say we would like to have an array to icons like numbers, to order the elements of my carousel numerically. Let's do this!
 
```jsx
const anArrayOfNumbers = [<img src="http://random.com/one"/>, 
                          <img src="http://random.com/two"/>, 
                          <img src="http://random.com/three"/>
                         ];

<Carousel
    IndicatorIcon={anArrayOfNumbers}
>
    {...}
</Carousel>
```

#### Example #3

Now we want to do more complex customizations. Specifically:

1. More distance between the indicator icons
2. Change the background color of the active indicator to `red`
3. Change the color of all indicators to `blue`
4. Move the indicators to the right side of the carousel
5. Move the indicators to be further away down from the carousel

We are going to use all props to style the indicators

```jsx
import Home from '@mui/icons-material/Home';

<Carousel
    IndicatorIcon={<Home/>} // Previous Example
    indicatorIconButtonProps={{
        style: {
            padding: '10px',    // 1
            color: 'blue'       // 3
        }
    }}
    activeIndicatorIconButtonProps={{
        style: {
            backgroundColor: 'red' // 2
        }
    }}
    indicatorContainerProps={{
        style: {
            marginTop: '50px', // 5
            textAlign: 'right' // 4
        }

    }}
>
    {...}
</Carousel>
```

As before, you can use `className` to style the elements externally.

### Default Styles

Giving the default styles in pseudo-code.

#### Navigation Buttons

```js
{
    buttonWrapper: {
        position: "absolute",
        height: "100px",
        backgroundColor: "transparent",
        top: "calc(50% - 70px)",
        '&:hover': {
            '& $button': {
                backgroundColor: "black",
                filter: "brightness(120%)",
                opacity: "0.4"
            }
        }
    },
    fullHeightHoverWrapper: {
        height: "100%",
        top: "0"
    },
    buttonVisible:{
        opacity: "1"
    },
    buttonHidden:{
        opacity: "0",
    },
    button: {
        margin: "0 10px",
        position: "relative",
        backgroundColor: "#494949",
        top: "calc(50% - 20px) !important",
        color: "white",
        fontSize: "30px",
        transition: "200ms",
        cursor: "pointer",
        '&:hover': {
            opacity: "0.6 !important"
        },
    },
    // Applies to the "next" button wrapper
    next: {
        right: 0
    },
    // Applies to the "prev" button wrapper
    prev: {
        left: 0
    }
}
```

#### Indicators

```js
{
    indicators: {
        width: "100%",
        marginTop: "10px",
        textAlign: "center"
    },
    indicator: {
        cursor: "pointer",
        transition: "200ms",
        padding: 0,
        color: "#afafaf",
        '&:hover': {
            color: "#1f1f1f"
        },
        '&:active': {
            color: "#1f1f1f"
        }
    },
    indicatorIcon: {
        fontSize: "15px",
    },
    // Applies to the active indicator
    active: {           
        color: "#494949"
    }
}
```

## Props

| Prop name                      | Type                                                                                                                                                       | Default                                                                   | Description                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
| ------------------------------ | ---------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| sx                             | `SxProps<Theme>`                                                                                                                                           | `{}`                                                                      | Defines `sx` props, that will be **inserted** into the Carousel 's root element                                                                                                                                                                                                                                                                                                                                                                                                                                                |
| className                      | `string`                                                                                                                                                   | ""                                                                        | Defines custom class name(s), that will be **added** to Carousel element                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
| height                         | `number \| string`                                                                                                                                         | `undefined`                                                               | Defines the carousel's height in `px`. If this is not set, the carousel's height will be the height of its children. If it is not set, the carousel's height will be the same as the current active child.                                                                                                                                                                                                                                                                                                                                                                                                            |
| index                          | `number`                                                                                                                                                   | `0`                                                                       | Defines which child (assuming there are more than 1 children) will be displayed. Next and Previous Buttons as well as Indicators will work normally after the first render. When this prop is updated the carousel will display the chosen child. *Use this prop to programmatically set the active child*. If (index > children.length) then if (strictIndexing) index = last element. index                                                                                                                                  |
| strictIndexing                 | `boolean`                                                                                                                                                  | `true`                                                                    | Defines whether index can be bigger than children length                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
| autoPlay                       | `boolean`                                                                                                                                                  | `true`                                                                    | Defines if the component will auto scroll between children                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
| stopAutoPlayOnHover            | `boolean`                                                                                                                                                  | `true`                                                                    | Defines if auto scrolling will continue while mousing over carousel                                                                                                                                                                                                                                                                                                                                                                                                                                                            |
| interval                       | `number`                                                                                                                                                   | `4000`                                                                    | Defines the interval in **ms** between active child changes (autoPlay)                                                                                                                                                                                                                                                                                                                                                                                                                                                         |
| animation                      | `"fade"  \| "slide"`                                                                                                                                       | `"fade"`                                                                  | Defines the animation style of the Carousel                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
| duration                       | `number`                                                                                                                                                   | `500`                                                                     | Defines the duration of the animations.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        |
| swipe                          | `boolean`                                                                                                                                                  | `true`                                                                    | Defines if swiping left and right (in touch devices) triggers `next` and `prev` behaviour                                                                                                                                                                                                                                                                                                                                                                                                                                      |

CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

The format is based on [Keep a Changelog][Keep a Changelog] and this project adheres to [Semantic Versioning][Semantic Versioning].

## [3.4.0 - 3.4.2] - *May 28 2022*

### Fixed

* Fixes [#issue 173](https://github.com/Learus/react-material-ui-carousel/issues/173). Problem was caused by framer-motion enter animation div having 0 `height`, and hence making the carousel have 0 `height`, when the animation took a long time to start, or the user was spamming the next button. Now, the library checks for 0 `height`, and retries to set `height` after a few ms until the div's `offsetHeight` isn't 0 anymore. ([@learus])

### Added

* The carousel now supports React 18 as well as React 17. ([@aveloso4], [@learus] )

## [3.3.0 - 3.3.3] - *Mar 25 2022*

### Changed

* The carousel now changes its height to be exactly the height of the current active child.

### Added

* Added the `height` prop, to specify an exact height for the carousel. If it is specified, the carousel's height will no longer be the active child's height.
* Added a section in the demo to demonstrate dynamic height functionality.

### Fixed

* :bug: Fixed the `index` bug, where the carousel wouldn't update when the `index` prop was changed.
* Removed unnecessary and forgotten console logs

## [3.2.0] - *Feb 10 2022*

### Added

* Added support for sx prop on Carousel container ([@matyas-igor])
* Now `autoPlay` is paused when carousel is focused using keyboard, hence making the library more accessible. [Issue.](https://github.com/Learus/react-material-ui-carousel/issues/137) ([@hajineats])
* Links to contributor profiles on CHANGELOG.md

### Changed

* Converted makeStyles into using styled helper ([@matyas-igor])
* Updated the file structure to be more readable ([@Learus])
* **(Breaking)** Added `@mui/system` as peer dependency.  
  * *Reason*: The newly added `sx` prop is of type `SxProps` imported from `@mui/system`.

### Removed

* Removed legacy @mui/styles (and thus jss dependency) ([@matyas-igor])

## [3.1.2] - *Jan 22 2021*

This version is unpublished. Its changes are moving to version 3.2.0 among others.

## [3.0.2] - *Oct 29 2021*

### Changed

* Changed `next`, `prev`, and `onChange` typings to reflect their arguments and return value.

## [3.0.0] - *Oct 28 2021*

:fireworks::fire: React Material Ui Carousel has been fully refactored to use more modern react techniques such as hooks, functional components and other cool stuff. :fire::fireworks:  
In general, its usage does not change.

### Changed

* The library is now written in Typescript, for easier development, debugging and smaller package sizing.
* **(Breaking)** Now supports MUI 5 **by default**. Version 3 will not work with MUI 4 as of now. ([@Learus])
  * Hence, minimum React version has been bumped to `^17.0.1`
* **(Breaking)** Now uses [`framer/motion`](https://github.com/framer/motion) to handle animations and gesture support. More animations will be added in the future. ([@Learus])
  * "Enter" and "Exit" animations are now simultaneous. No flash/gap of background between slide changes. Fixes issues [20](https://github.com/Learus/react-material-ui-carousel/issues/20), [61](https://github.com/Learus/react-material-ui-carousel/issues/61), [66](https://github.com/Learus/react-material-ui-carousel/issues/66).
  * The first carousel item now renders without an "initial" animation. Implements feature request [92](https://github.com/Learus/react-material-ui-carousel/issues/92).
  * **(Breaking)** The `timeout` prop is now `duration` and only supports a number that handles every aspect of the animation (enter, exit, e.t.c.)
* **(Breaking)** Now renders (preloads) all items initially **and does not rerender** them, unless there has been a state change. Implements feature request [88](https://github.com/Learus/react-material-ui-carousel/issues/88) ([@Learus])
* **(Breaking)** `CarouselStyleProps` &#8594; `CarouselNavProps` ([@Learus])
* Updated the [Demo page](https://learus.github.io/react-material-ui-carousel) look.

### Fixed

* `findDOMNode is deprecated in StrictMode.` warning has now been fixed. Material UI animations are no longer being used. [Issue.](https://github.com/Learus/react-material-ui-carousel/issues/32) ([@Learus])
* `Can't perform a React state update on an unmounted component.` warning has now been fixed. [Issue.](https://github.com/Learus/react-material-ui-carousel/issues/44) ([@Learus])

### Removed

* Removed any dependencies that were necessary for version 2. Only dependency now is `framer/motion`

## [2.3.5] - *Sep 22 2021*

### Fixed

- Added fullHeightHoverButton class to styles, to fix `property does not exist` error. [Issue.](https://github.com/Learus/react-material-ui-carousel/issues/110) ([@Learus])

### Changed

- Removed unnecessary ternary operators. [Issue.](https://github.com/Learus/react-material-ui-carousel/issues/111) ([@Learus])

## [2.3.1] - *August 15 2021*

### Fixed

- Fixed author link on README.md. It didn't work. (@hamidreza-nateghi)

## [2.3.0] - *August 15 2021*

### Added

- Added all `aria` props to:
  - `navButtonsWrapperProps`
  - `navButtonsProps`
  - `indicatorIconButtonProps`
  - `activeIndicatorIconButtonProps`
  - `indicatorContainerProps` ([@Learus])
- Added `reverseEdgeAnimationDirection` prop. Normally, the Carousel reverses the animation direction when going from its last child to its first (right --> left). It also reverses the animation direction when going from its first child to its last (left --> right). This prop disables this functionality.
  Now every "next" button press will have a right directed animation and each "previous" button press will have a left directed animation. ([@Learus])

## [2.2.7] - *July 5 2021*

### Fixed

- Fixed a bug where it was necessary to provide both `className` and `style` when styling buttons or indicators, when using typescript. ([@Learus])

## [2.2.6] - *May 7 2021*

### Changed


components/Carousel.js
'use strict';

Object.defineProperty(exports, "__esModule", {
    value: true
});

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _core = require('@material-ui/core');

var _autoBind = require('auto-bind');

var _autoBind2 = _interopRequireDefault(_autoBind);

var _FiberManualRecord = require('@material-ui/icons/FiberManualRecord');

var _FiberManualRecord2 = _interopRequireDefault(_FiberManualRecord);

var _NavigateBefore = require('@material-ui/icons/NavigateBefore');

var _NavigateBefore2 = _interopRequireDefault(_NavigateBefore);

var _NavigateNext = require('@material-ui/icons/NavigateNext');

var _NavigateNext2 = _interopRequireDefault(_NavigateNext);

require('../style/Carousel.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Carousel = function (_Component) {
    _inherits(Carousel, _Component);

    function Carousel(props) {
        _classCallCheck(this, Carousel);

        var _this = _possibleConstructorReturn(this, (Carousel.__proto__ || Object.getPrototypeOf(Carousel)).call(this, props));

        _this.state = {
            active: 0,
            autoPlay: _this.props.autoPlay !== undefined ? _this.props.autoPlay : true,
            interval: _this.props.interval !== undefined ? _this.props.interval : 4000
        };

        _this.timer = null;

        (0, _autoBind2.default)(_this);
        return _this;
    }

    _createClass(Carousel, [{
        key: 'componentDidMount',
        value: function componentDidMount() {
            this.start();
        }
    }, {
        key: 'componentWillUnmount',
        value: function componentWillUnmount() {
            this.stop();
        }
    }, {
        key: 'componentDidUpdate',
        value: function componentDidUpdate(prevProps, prevState) {
            if (prevProps.autoPlay !== prevState.autoPlay || prevProps.interval !== prevState.interval) {
                this.reset();
            }
        }
    }, {
        key: 'stop',
        value: function stop() {
            if (this.timer) {
                clearInterval(this.timer);
                this.timer = null;
            }
        }
    }, {
        key: 'start',
        value: function start() {
            if (this.state.autoPlay) {
                this.timer = setInterval(this.next, this.state.interval);
            }
        }
    }, {
        key: 'reset',
        value: function reset() {
            this.stop();

            if (this.state.autoPlay) {
                this.start();
            }
        }
    }, {
        key: 'pressIndicator',
        value: function pressIndicator(index) {
            this.setState({
                active: index
            }, this.reset);
        }
    }, {
        key: 'next',
        value: function next(event) {
            var next = this.state.active + 1 > this.props.children.length - 1 ? 0 : this.state.active + 1;

            this.setState({
                active: next
            }, this.reset);

            if (event) event.stopPropagation();
        }
    }, {
        key: 'prev',
        value: function prev(event) {
            var prev = this.state.active - 1 < 0 ? this.props.children.length - 1 : this.state.active - 1;

            this.setState({
                active: prev
            }, this.reset);

            if (event) event.stopPropagation();
        }
    }, {
        key: 'render',
        value: function render() {
            var _this2 = this;

            var indicators = this.props.indicators !== undefined ? this.props.indicators : true;
            var animation = this.props.animation !== undefined ? this.props.animation : "fade";

            return _react2.default.createElement(
                'div',
                { className: 'Carousel ' + (this.props.className ? this.props.className : ""), onMouseEnter: this.stop, onMouseOut: this.reset },
                Array.isArray(this.props.children) ? this.props.children.map(function (child, index) {
                    return _react2.default.createElement(CarouselItem, { key: index, active: index === _this2.state.active ? true : false, child: child, animation: animation });
                }) : _react2.default.createElement(CarouselItem, { key: 0, active: true, child: this.props.children }),
                _react2.default.createElement(
                    'div',
                    { className: 'Next ButtonWrapper' },
                    _react2.default.createElement(
                        _core.IconButton,
                        { className: 'Next Button mui--align-middle', onClick: this.next },
                        _react2.default.createElement(_NavigateNext2.default, null)
                    )
                ),
                _react2.default.createElement(
                    'div',
                    { className: 'Prev ButtonWrapper' },
                    _react2.default.createElement(
                        _core.IconButton,
                        { className: 'Prev Button mui--align-middle', onClick: this.prev },
                        _react2.default.createElement(_NavigateBefore2.default, null)
                    )
                ),
                indicators ? _react2.default.createElement(Indicators, { length: this.props.children.length, active: this.state.active, press: this.pressIndicator }) : null
            );
        }
    }], [{
        key: 'getDerivedStateFromProps',
        value: function getDerivedStateFromProps(nextProps, prevState) {
            if (nextProps.autoPlay !== prevState.autoPlay || nextProps.interval !== prevState.interval) {
                return {
                    autoPlay: nextProps.autoPlay !== undefined ? nextProps.autoPlay : true,
                    interval: nextProps.interval !== undefined ? nextProps.interval : 4000
                };
            } else return null;
        }
    }]);

    return Carousel;
}(_react.Component);

exports.default = Carousel;


function CarouselItem(props) {
    return (
        // props.active ? 
        // (
        _react2.default.createElement(
            'div',
            { className: 'CarouselItem', hidden: !props.active },
            props.animation === "slide" ? _react2.default.createElement(
                _core.Slide,
                { direction: 'left', 'in': props.active, timeout: 200 },
                _react2.default.createElement(
                    'div',
                    null,
                    props.child
                )
            ) : _react2.default.createElement(
                _core.Fade,
                { 'in': props.active, timeout: 500 },
                _react2.default.createElement(
                    'div',
                    null,
                    props.child
                )
            )
        )
        // ) : null

    );
}

function Indicators(props) {
    var indicators = [];

    var _loop = function _loop(i) {
        var className = i === props.active ? "Active Indicator" : "Indicator";
        var item = _react2.default.createElement(_FiberManualRecord2.default, { key: i, size: 'small', className: className, onClick: function onClick() {
                props.press(i);
            } });

        indicators.push(item);
    };

    for (var i = 0; i < props.length; i++) {
        _loop(i);
    }

    return _react2.default.createElement(
        'div',
        { className: 'Indicators' },
        indicators
    );
}